client/src/main/java/io/split/client/SplitFactoryBuilder.java

```diff
@@ -113,7 +116,18 @@ public static synchronized SplitFactory build(String apiToken, SplitClientConfig
         final RefreshableSplitFetcherProvider splitFetcherProvider = new RefreshableSplitFetcherProvider(splitChangeFetcher, splitParser, findPollingPeriod(RANDOM, config.featuresRefreshRate()), gates);
 
         // Impressions
-        final TreatmentLog treatmentLog = ImpressionsManager.instance(httpclient, config);
+        final ImpressionsManager splitImpressionListener = ImpressionsManager.instance(httpclient, config);
+        final ImpressionListener impressionListener;
+
+        if (config.impressionListener() != null) {
+            AsynchronousImpressionListener wrapper = AsynchronousImpressionListener.build(config.impressionListener(), config.impressionListenerCapactity());
```

Comment:
when both listeners exist, then we create a federated listener of our listener (that uploads impressions to our endpoint) and theirs